recipes/ohnet/all/conanfile.py

```diff
@@ -0,0 +1,145 @@
+from conan import ConanFile
+from conan.tools.files import get, chdir, copy, mkdir, export_conandata_patches, apply_conandata_patches
+from conan.tools.apple import fix_apple_shared_install_name, is_apple_os
+from conan.tools.gnu import Autotools, AutotoolsToolchain
+from conan.tools.microsoft import is_msvc, msvc_runtime_flag, NMakeToolchain
+from conan.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.53.0"
+
+
+class OhNetConan(ConanFile):
+    name = "ohnet"
+    description = "OpenHome Networking (ohNet) is a modern, cross platform, multi-language UPnP stack"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/openhome/ohNet"
+    topics = ("openhome", "ohnet", "upnp")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    short_paths = True
+
+    def _get_openhome_architecture(self, args):
+        if is_apple_os(self):
+            if str(self.settings.arch).startswith("armv8"):
+                openhome_architecture = "arm64"
+                args.extend([f"openhome_architecture={openhome_architecture}", f"detected_openhome_architecture={openhome_architecture}"])
+        return args
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
```

Comment:
```suggestion

    def layout(self):
        basic_layout(self, src_folder="src")

```

The advantage using layout is not mixing source files with build cache in the build folder, so when developing a package, you can list/run much easier in case an error.

https://docs.conan.io/2/tutorial/developing_packages/package_layout.html